File: Guanabara - POO Java/Meus exercícios POO/Aula5 - Exercicio Conta Bancaria/src/main/java/aula5/ContaBanco.java
package aula5;
public class ContaBanco{
//declarando variáveis
    public int numConta;
    protected String tipo; //`cc` ou `cp`
    private String dono;
    private float saldo;
    private boolean status; //aberta ou fechada

//método construtor
    public ContaBanco(int numConta, String dono) {        
        this.status = false; //atributos pré-valorados
        this.saldo = 0f;
        
        this.numConta = numConta; //atributos valorados pelos parâmetros passados na criação do objeto
        this.dono = dono;
    }
    
//métodos acessores
    public int getNumConta() {
        return this.numConta;
    }

    public void setNumConta(int nc) {
        this.numConta = nc;
    }

    public String getTipo() {
        return this.tipo;
    }

    public void setTipo(String t) { //essa variável `t` pertence apenas ao escopo LOCAL, por isso whatever seu nome
        this.tipo = t;       
    }

    public String getDono() {
        return this.dono;
    }

    public void setDono(String d) {
        this.dono = d;
    }

    public float getSaldo() {
        return this.saldo;
    }

    public void setSaldo(float s) {
        this.saldo = s;
    }

    public boolean getStatus() {
        return this.status;
    }

    public void setStatus(boolean s) {
        this.status = s;
    }
    
//métodos públicos da classe    
    public void informacoes() {
        System.out.println("DADOS DO CLIENTE:");
        System.out.println("Número da conta: " + this.getNumConta());
        System.out.println("Tipo: " + this.getTipo());
        System.out.println("Dono: " + this.getDono());
        System.out.println("Status: " + this.getStatus());
        System.out.println("Saldo: " + this.getSaldo());
    }
    public void abrirConta(String t){     
        
        if (t.equals("cc") || t.equals("cp")) {
            this.setTipo(t);
            this.setStatus(true);
            System.out.println("Conta aberta com sucesso!");
        } else {
            System.out.println("[ERRO] Esse tipo de conta não existe!");
        }
        //Bônus para novos clientes
        if (this.getTipo().equals("cc")) {
            depositar(50f);
        } else if (this.getTipo().equals("cp")) {
            depositar(150f);
        }
    }
    
    public void fecharConta(){
        if (this.getSaldo() > 0) {
                System.out.println("[ERRO] Conta com dinheiro! Sacando R$" + this.getSaldo());                
                sacar(this.getSaldo());
        } else if (this.getSaldo() < 0) {
                System.out.println("[ERRO] Conta em débito. Deposite R$" + Math.abs(this.getSaldo()));//Math.abs() gera valor absoluto, ou seja, positivo
        } else { //saldo zerado.
            System.out.println("Conta Encerrada");
        }
    }
    
    public void depositar(float v){
        if (this.getStatus() == true) { //verifica se a conta está ativa
            this.setSaldo(this.getSaldo() + v); //this.saldo = this.saldo += v (essa é uma maneira insegura)
            System.out.println("O depósito de " + v + " foi realizado na conta de " + this.getDono()); 
        } else {
            System.out.println("[ERRO] Impossível realizar a operação porque a conta está fechada.");
        }
    }
    
    public void sacar(float v){
        if (this.getStatus()) { //se tirar o trecho redundante: (== true), o programa entende mesmo assim como verdadeiro
            if (this.getSaldo() >= v ) {
                this.setSaldo(this.getSaldo() - v); //ou... saldo -= v; 
                System.out.println("Saque de R$" + v + " realizado na conta de " + this.getDono()); 
            } else {
                System.out.println("[ERRO] Saldo insuficiente para saque na conta de " + this.getDono());
            }            
        } else {
            System.out.println("[ERRO] Impossível sacar de uma conta fechada!");
        }
    }
    
    public void pagarMensal(){
        if (this.getStatus() == true) {
            if (this.getTipo().equals("cc") && this.getSaldo() > 12) {
                sacar(12f);
                System.out.println("Mensalidade paga com sucesso por " + this.getDono());
            } else if (this.getTipo().equals("cp") && this.getSaldo() > 20) {
                sacar(20f);
                System.out.println("Mensalidade paga com sucesso por " + this.getDono());
            } else {
                System.out.println("Saldo insuficiente para pagar a mensalidade!");
            }             
        } else {
            System.out.println("[ERRO] Conta está fechada! Impossível realizar a operação.");
        }
    }
}

File: PHP/exercicios-PHPOO/Aula05-ProjetoBanco/ContaBanco.php
<?php
class ContaBanco {
//Atributos
public $NumConta;
protected $tipo;
private $dono;
private $saldo;
private $status;

//Métodos
public function abrirConta($t){
    $this->setTipo($t);
    $this->setStatus(true);
    if ($t == "CC") {
        $this->setSaldo(50);
    } elseif ($t == "CP") {
        $this->setSaldo(150);
    }
    
}
public function fecharConta(){
    if ($this->getSaldo() > 0) {
        echo "<p>Conta ainda tem dinheiro, não posso fechá-la!</p>";
    } elseif ($this->getSaldo() < 0) {
        echo "<p>Conta possui débitos. Impossível encerrar!</p>";
    } else {
        $this->setStatus(false);
        echo "<p>A conta de " . $this->getDono() . " foi encerrada com sucesso!</p>";
    }
}
public function depositar($v){
    if ($this->getStatus()) {
        $this->setSaldo($this->getSaldo() + $v);
      //$this->saldo += $v;   //sem o método acessor
      echo "<p>Depósito de R$ {$v} autorizado na conta de " . $this->getDono() . "</p>";
    } else {
        echo "<p>Conta fechada. Impossível depositar.</p>";
    }
}
public function sacar($v){
    if ($this->getStatus()) {
        if ($this->getSaldo() >= $v) {
            $this->setSaldo($this->getSaldo() - $v);
            echo "<p>Saque no valor de R$ {$v} autorizado na conta de " . $this->getDono() . "</p>"; //usando concatenação
        } else {
            echo "<p>Saldo insuficiente para esse saque.</p>";
        }
    } else {
        echo "<p>Não posso sacar de uma conta fechada.</p>";
    }
}
public function pagarMensalidade(){
    if ($this->getTipo() == "CC") {
        $v = 12;
    } elseif ($this->getTipo() == "CP") {
        $v = 20;
    }
    if ($this->getStatus()) {
        $this->setSaldo($this->getSaldo() - $v);
        echo "<p>Mensalidade de R$ {$v} debitada na conta de " . $this->getDono() . "</p>";
    } else {
        echo "<p>Problemas com a conta. Não posso cobrar.</p>";
    }
}

//Métodos Especiais
public function __construct()
{
    $this->setSaldo(0);
    $this->setStatus(false);
    echo "<p>Conta criada com sucesso!</p>";
}
public function getNumConta(){
    return $this->NumConta;
}
public function setNumConta($num){
    $this->NumConta = $num;
}
public function getTipo(){
    return $this->tipo;
}
public function setTipo($tip){
    $this->tipo = $tip;
}
public function getDono(){
    return $this->dono;
}
public function setDono($dono){
    $this->dono = $dono;
}
public function getSaldo(){
    return $this->saldo;
}
public function setSaldo($saldo){
    $this->saldo = $saldo;
}
public function getStatus(){
    return $this->status;
}
public function setStatus($status){
    $this->status = $status;
}
}